### ISSUE TITLE
Throw an error when callback is provided (or document change)

### ISSUE THREAD

**Issue by JedWatson** [sindresorhus/del#36]
I realise this is totally my fault (because there's a major version bump) and you may not want the extra code in there, but I got caught by the switch from callbacks to promises when updating a bunch of dependencies.

Since there's no Changelog, a warning the API has changed would go a long way to making upgrades more painless; failing that, explicitly calling out the switch from callbacks to Promises near the top of the README would also help. As it is you need to read the docs and realise there's no longer a reference to callback support anywhere, and check the commit history to figure out what's going on.

**Comment by arthurvr** [comment]
We do our best to keep a clean commit log so for tiny modules like this a changelogs every release or deprecation warnings aren't really needed. Between 1.2.1 and 2.0 there were literally 3 commits so it's very easy to find out the change ;)

**Comment by sindresorhus** [comment]
https://github.com/sindresorhus/del/releases/tag/v2.0.0

**Comment by JedWatson** [comment]
> https://github.com/sindresorhus/del/releases/tag/v2.0.0

right, that's totally what I was after :)

will check the releases section next time! is that worth a link in the Readme? I'm used to looking for a CHANGELOG.md.

either way, thanks!

**Comment by sindresorhus** [comment]
No, that's expected knowledge.

**Comment by JedWatson** [comment]
> No, that's expected knowledge.

Unfortunately many package maintainers ignore it. npm & react use it; node and babel don't.

When you're upgrading modules, it's really nice to have something front and centre on the [npm package page](https://www.npmjs.com/package/del) that says where to go to find whatever you need to know to upgrade. Instead you have to explore to figure out which one's going to be most helpful; CHANGELOG.md in the repo, GitHub releases, or the commit history. Sometimes there's a post on a website somewhere. So IMO it's a worthwhile addition to the Readme.

Anyway, not here to rant at you, maybe this is something better solved by npm with an addition to package.json: npm/npm#9740

**Comment by sindresorhus** [comment]
I meant, it's expected knowledge to check GitHub Releases if there's no changelog.md.

> CHANGELOG.md in the repo, GitHub releases, or the commit history

Sounds like something a tool should solve, though, not every single readme out there. `$ npm update --list-changes` or even better, a separate CLI tool.

**Comment by brendanfalkowski** [comment]
I was just in the same position as @JedWatson. I've only used Node for Gulp / frontend stuff, and didn't know releases for changelog info (I came straight to issues).

I really appreciate the work that goes into maintaining these packages, and know a line must be drawn (here...no further!) on expected knowledge. Now I know to check releases too. That little example helped me solve my broken build right away.
